vnext/Microsoft.ReactNative.Cxx/JSValue.cpp

```diff
@@ -218,132 +887,29 @@ bool JSValue::Equals(const JSValue &other) const noexcept {
     case JSValueType::Object:
-      WriteObjectTo(writer, m_object);
-      break;
+      return m_object.WriteTo(writer);
     case JSValueType::Array:
-      WriteArrayTo(writer, m_array);
-      break;
+      return m_array.WriteTo(writer);
     case JSValueType::String:
-      writer.WriteString(to_hstring(m_string));
-      break;
+      return writer.WriteString(to_hstring(m_string));
     case JSValueType::Boolean:
-      writer.WriteBoolean(m_bool);
-      break;
+      return writer.WriteBoolean(m_bool);
     case JSValueType::Int64:
-      writer.WriteInt64(m_int64);
-      break;
+      return writer.WriteInt64(m_int64);
     case JSValueType::Double:
-      writer.WriteDouble(m_double);
-      break;
-  }
-}
-
-/*static*/ void JSValue::WriteObjectTo(IJSValueWriter const &writer, JSValueObjectView object) noexcept {
-  writer.WriteObjectBegin();
-  for (const auto &property : object) {
-    writer.WritePropertyName(to_hstring(property.first));
-    property.second.WriteTo(writer);
-  }
-  writer.WriteObjectEnd();
-}
-
-/*static*/ void JSValue::WriteArrayTo(IJSValueWriter const &writer, JSValueArrayView value) noexcept {
-  writer.WriteArrayBegin();
-  for (const JSValue &item : value) {
-    item.WriteTo(writer);
-  }
-  writer.WriteArrayEnd();
-}
-
-bool JSValue::ObjectEquals(const JSValueObject &other) const noexcept {
-  if (m_object.size() != other.size()) {
-    return false;
-  }
-
-  for (const auto &entry : m_object) {
-    auto it = other.find(entry.first);
-    if (it == other.end()) {
-      return false;
-    }
-
-    if (!entry.second.Equals(it->second)) {
-      return false;
-    }
-  }
-
-  return true;
-}
-
-bool JSValue::ArrayEquals(const JSValueArray &other) const noexcept {
-  if (m_array.size() != other.size()) {
-    return false;
-  }
-
-  auto otherIt = other.begin();
-  for (const JSValue &item : m_array) {
-    if (!item.Equals(*(otherIt++))) {
-      return false;
-    }
-  }
-
-  return true;
-}
-
-// Read object properties without checking value type.
-/*static*/ JSValueObject JSValue::ReadObjectProperties(IJSValueReader const &reader) noexcept {
-  JSValueObject properties;
-  hstring propertyName;
-  while (reader.GetNextObjectProperty(/*ref*/ propertyName)) {
-    properties.emplace(to_string(propertyName), ReadFrom(reader));
-  }
-
-  return properties;
-}
-
-// Read array items without checking value type.
-/*static*/ JSValueArray JSValue::ReadArrayItems(IJSValueReader const &reader) noexcept {
-  JSValueArray items;
-  while (reader.GetNextArrayItem()) {
-    items.push_back(ReadFrom(reader));
+      return writer.WriteDouble(m_double);
+    default:
+      VerifyElseCrashSz(false, "Unexpected JSValue type");
   }
-
-  return items;
-}
-
-//===========================================================================
-// Standalone functions implementation
-//===========================================================================
-
-void swap(JSValue &left, JSValue &right) noexcept {
```

Comment:
>swap [](start = 5, length = 4)

This implementation is no different from the default std::swap and thus it was removed.